examples/usart.rs

```diff
@@ -0,0 +1,21 @@
+#![no_std]
+#![no_main]
+#![feature(lang_items, core, collections, no_std)]
+
+extern crate core;
+extern crate emlib;
+#[macro_use]
+extern crate collections;
+
+use core::default::Default;
+use emlib::modules::Usart;
+
+#[no_mangle]
+pub extern fn main() {
+    let usart: Usart = Default::default();
+
+    loop {
+        let s = format!("Received: {}\n\r", usart.read_string());
+        usart.write_str(&s[..]);
```

Comment:
Don't know. I think this is better left as a task for later either way. 
I think we would have to implement some kind of io-trait, but it would probably also require the library to both _find and instantiate_ a Usart. The default Usart? Would println! only instantiate it once? How would it get access to the Usart?